CMakeLists.txt

```diff
@@ -1804,7 +1804,8 @@ if(CONFIG_BUILD_OUTPUT_INFO_HEADER)
     )
 endif()
 
-if(CONFIG_CHECK_INIT_PRIORITIES)
+if(CONFIG_CHECK_INIT_PRIORITIES AND
+   NOT CMAKE_C_COMPILER_ID STREQUAL "ARMClang")
```

Comment:
we try to make the general CMake build system as independent on compilers / os / etc as possible.

Toolchain specifics should try to be kept in dedicated cmake files for such purposes, such as those under `cmake/toolchain` / `cmake/compiler` so rest of CMake code is as common as possible.

Therefore better to ensure the Kconfig option is not available to users using such toolchain (see other comment).

Reply:
Sounds good, would have been my option as well but somehow I missed the Kconfig option when I looked for it, dropped the check here, but left it on the target as that's independent from the option.